### PR TITLE
I GIVE UP

### PR DESCRIPTION
**Pull request by mdcmeridius**
https://github.com/digitalbitbox/bitbox-wallet-app/issues/1774

Bitbox app settings (note: outdated screenshots, will be "Custom **Bitcoin** block explorer"):
image
image

Before clicking on transaction link:
image

After clicking on transaction link:
image

### CODE REVIEW

**Comment by mdcmeridius** [comment]
NOTE: this PR also includes changes to compile_windows.bat, so it becomes more flexible and works with other Visual Studio flavors (Professional, Enterprise, etc.).

**Comment by thisconnect** [comment]
looks very nice 👍 

please open a separate PR for the VS support.

https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/CONTRIBUTING.md

File: backend/config/config.go [hunk 1]
```diff
@@ -59,9 +59,15 @@ type proxyConfig struct {
...
+type blockExplorerConfig struct {
+	Enabled bool   `json:"useCustomBlockExplorer"`
+	URL     string `json:"blockExplorerURL"`
+}
+
 // Backend holds the backend specific configuration.
 type Backend struct {
-	Proxy proxyConfig `json:"proxy"`
+	Proxy            proxyConfig         `json:"proxy"`
+	BTCBlockExplorer blockExplorerConfig `json:"blockExplorer"`
```
Comment (benma) [inline backend/config/config.go:70]: please move this to `btcCoinConfig`, which allows extending this feature to testnet and alts in the future.

File: frontends/qt/compile_windows.bat [hunk 2]
```diff
@@ -1,6 +1,22 @@
 :: Compiles the Qt5 app. Part of `make windows`, which also compiles/bundles the deps
 
-call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"
+setlocal
```
Comment (benma) [inline frontends/qt/compile_windows.bat:3]: Please remove the build commits from this PR, as you moved it to #1784

File: frontends/web/src/routes/settings/settings.tsx [hunk 3]
```diff
@@ -93,6 +97,17 @@ class Settings extends Component<Props, State> {
...
   };
 
+  private handleBlkExpFormChange = (event: React.SyntheticEvent) => {
+    const target = (event.target as HTMLInputElement);
+    if (target.name !== 'blockExplorerURL') {
+      return;
+    }
+    this.setState({
+      [target.name]: target.value,
+      restart: false,
```
Comment (benma) [inline frontends/web/src/routes/settings/settings.tsx:107]: remove this? if the restart flag is set it means a restart is needed. Changing the block explorer form does not change that.

**Review by benma — CHANGES_REQUESTED** [review]

File: backend/config/config.go [hunk 4]
```diff
@@ -59,9 +59,15 @@ type proxyConfig struct {
 	ProxyAddress string `json:"proxyAddress"`
 }
 
+type blockExplorerConfig struct {
+	Enabled bool   `json:"useCustomBlockExplorer"`
+	URL     string `json:"blockExplorerURL"`
```
[the remark this answers is not in the document]
Comment (mdcmeridius) [reply backend/config/config.go:64]: I agree.

**Comment by mdcmeridius** [comment]
I'm giving this PR because the maintainers of this project are too picky and not very smart.
If you want a sucky product, suit yourselves!
GOODBYE

**Pull request closed without merging**